### PR TITLE
feat: Discussion Post with Replies

### PR DESCRIPTION
**Pull request by calebpanza**
## ✏️ Description
This PR is a followup to #2929 which allows a user to select a Thread of a Discussion Post and reply to the thread with their own contribution

In the PR, a User can:
1. View a paginated thread of replies to a discussion post
2. Users can add a text-only reply
3. Replies will automatically refresh upon submitting a reply
4. iOS and Android safe areas/keyboard avoid have been taken into account
5. Pull-to-refresh

## 🔬 To Test
1. Open up the Communities Tab
2. Tap on the "reply" call to action on a Discussion Post
3. View and add Replies

## 📸 Screenshots
| Thread |
| --- |
| Screenshot_20230201-123842 |

## ⚠️ To-do Before Merge

- [ ] Ensure PR #2929 is merged

## 📝 Checklist

- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)
- [x] For PR's with design updates/changes, I have consulted with the product designer and made all necessary adjustments in Figma

### CODE REVIEW

**Comment by dsilva609** [comment]
@calebpanza would it be helpful to also have the "time ago" on replies as well? Kind of hard to tell what order things are in at first glance

**Comment by dsilva609** [comment]
@calebpanza 
image

File: templates/mobile/src/components/activities/PostActivity/UserReplyBar.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import ReplyReaction from '../ReplyReaction';
+
+const UserReplyBar = ({}) => {
+  const navigation = useNavigation();
+  const { activityId, actorProfile, replies, numberOfReplies } =
+    useActivityContext();
+  const { profile } = useCurrentPersonProfile();
+  const [opacityStyles, onPressIn, onPressOut] = usePressableOpacityStyle();
+  const hasReplies = numberOfReplies > 0;
+  const shouldShowReplyAction = numberOfReplies < 2;
```
Comment (dsilva609) [inline templates/mobile/src/components/activities/PostActivity/UserReplyBar.js:38]: could probably pull this count to a constant

File: templates/mobile/src/screens/UserPostThread/UserPostThread.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  const disabled = loading || mutationLoading;
+
+  async function fetchMore() {
+    /**
+     * Because our feed is naturally ordered in descending order (newest replies to oldest), we have to make sure that we use the "less than" query in order to fetch "older" replies upon pagination
+     */
+    const reactions = await streamClient.reactions.filter({
+      activity_id: activityId,
+      kind: 'REPLY',
+      limit: 5,
```
Comment (dsilva609) [inline templates/mobile/src/screens/UserPostThread/UserPostThread.js:75]: would it help to pull this limit out to a constant?

File: templates/mobile/src/ui-kit/components/SocialEngagement/SocialEngagement.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ const SocialEngagement = ({ label, profiles }) => {
       {renderAvatars ? <AvatarLine size="xs" profiles={profiles} /> : null}
       {renderAvatars && renderLabel ? <Spacer /> : null}
       {renderLabel ? (
-        <BodyText small color="text.tertiary">
+        <BodyText small color="tex.tertiary">
```
Comment (dsilva609) [inline templates/mobile/src/ui-kit/components/SocialEngagement/SocialEngagement.js:29]: is this supposed to be `text.`?

**Comment by calebpanza** [comment]
> would it be helpful to also have the "time ago" on replies as well

I hear ya! It's not a part of the current designs in Figma, so I'm gonna leave it as is. But that kinda feedback would be great for a product retro

File: templates/mobile/src/ui-kit/components/SocialEngagement/SocialEngagement.js (hunk 3 above)
Comment (calebpanza) [reply templates/mobile/src/ui-kit/components/SocialEngagement/SocialEngagement.js]: Good catch!

File: templates/mobile/src/screens/UserPostThread/UserPostThread.js (hunk 2 above)
Comment (calebpanza) [reply templates/mobile/src/screens/UserPostThread/UserPostThread.js]: Found a few more places to introduce constants, too. Great callout!

**Comment by calebpanza** [comment]
@dsilva609 updated that Text Input

File: templates/mobile/src/index.js [hunk 4]
```diff
@@ -44,6 +44,8 @@ import {
   ScreenNames,
   ScriptureExperience,
   UserPostInput,
+  UserPostThread,
+  ScreenNames,
```
Comment (dsilva609) [inline templates/mobile/src/index.js:48]: this is declared twice

**Review by dsilva609 — APPROVED** [review]

**Pull request merged**